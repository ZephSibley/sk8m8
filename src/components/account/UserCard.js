import React, { useState, useEffect } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import UpdateStatusForm from '../ui/forms/UpdateStatusForm';

import spinner from '../../assets/img/ajax-loader.gif'
import UploadAvatarForm from '../ui/forms/UploadAvatarForm';


const UserCard = props => {
    // Prop: requests; http client

    const [userDetails, updateUserDetails] = useState({
        username: 'Loading',
        avatar: spinner,
        status: ''
    })

    useEffect(() => {
        // Get location, add to queryurl
        props.requests.get(`
            ${process.env.REACT_APP_ENDPOINT}
            /Account/Details
        `).then(data => 
            updateUserDetails(data)
        ).catch(err =>
            updateUserDetails({
                username: 'Something went wrong D:',
                avatar: '/',
                status: err.message
            })
        );          
    }, [props.requests])

    return (
        <Card>
            <CardHeader
                /*avatar={
                    userDetails.avatar ?
                    <Avatar
                        alt={userDetails.username}
                        src={userDetails.avatar}
                    />
                    :
                    <UploadAvatarForm requests={props.requests} />
                }*/ // AVATAR FEATURE
                
                // action={
                //     <IconButton aria-label="settings">
                //         <MoreVertIcon />
                //     </IconButton>
                // }
                title={userDetails.username}
                subheader="avatars coming soon"
            />
            <CardContent>
                <UpdateStatusForm 
                    currentStatus={userDetails.status} 
                    requests={props.requests}
                />
            </CardContent>
        </Card>
    );
}

export default UserCard;