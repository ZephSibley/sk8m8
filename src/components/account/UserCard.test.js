import React from 'react';
import renderer from 'react-test-renderer';
import UserCard from './UserCard';

test('Basic Rendering', () => {
    const component = renderer.create(
        <UserCard
            username='testymctestface'
            bio='hello, this is dog'
            avatar=''
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})