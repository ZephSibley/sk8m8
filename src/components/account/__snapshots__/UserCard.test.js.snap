// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root"
      >
        <img
          alt="Loading"
          className="MuiAvatar-img"
          src="ajax-loader.gif"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Loading
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <div>
      <form
        className="makeStyles-form-69"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            status
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
            onClick={[Function]}
          >
            <textarea
              aria-invalid={false}
              className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
              disabled={false}
              name="status"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Whatcha sayin?"
              required={false}
              rows="4"
              value=""
            />
          </div>
          
        </div>
        <span
          className="MuiFormHelperText-root Mui-error"
        >
          
        </span>
        <button
          className="MuiButtonBase-root MuiButton-root makeStyles-submit-71 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          margin="normal"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Update
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
